packages/uni_app/lib/view/home/widgets/schedule_card.dart

```diff
@@ -49,6 +50,13 @@ class ScheduleCard extends GenericCard {
         ),
       ),
       contentLoadingWidget: const ScheduleCardShimmer().build(context),
+      mapper: (lectures) => lectures
+          .where(
+            (lecture) =>
+                lecture.endTime.isAfter(DateTime.now()) ||
+                lecture.startTime.weekday != DateTime.now().weekday,
```

Comment:
Is this || condition needed? It came from the `getScheduleRows` method, but maybe it was useless there, no?

We are checking if a lecture starts on a different weekday or ends after now.
The condition of the different weekday would only be used if a class has already ended (because of the ||). So, do we want classes that have already ended, but are on another weekday? I don't think so... What do you think?

This might be leftovers from when we refactored the lecture fetching system. Previously, we didn't store the date of the lecture, we only stored the time and the weekday. This meant that, on a friday, if we wanted to show classes from monday, we would have to ensure that the weekday is different from friday's. This isn't how the lecture system works anymore.

Reply:
If my thinking above is correct, we can just use `(lecture) => lecture.endTime.isAfter(DateTime.now())`

Reply:
Yes, I tested and it seemed to work without the weekday comparison, but I thought it was there for a reason, so I left it, but I'm going to remove it.